x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,323 @@
+      )) as ActionTypeExecutorResult<CrowdstrikeGetAgentsResponse>;
+
+      this.log.debug(`Response for Crowdstrike agent id [${id}] returned:\n${stringify(response)}`);
+
+      crowdstrikeApiResponse = response.data;
+    } catch (err) {
+      throw new ResponseActionsClientError(
+        `Error while attempting to retrieve Crowdstrike host with agent id [${id}]`,
+        500,
+        err
+      );
+    }
+
+    if (!crowdstrikeApiResponse || !crowdstrikeApiResponse.resources[0]) {
+      throw new ResponseActionsClientError(`Crowdstrike agent id [${id}] not found`, 404);
+    }
+
+    return crowdstrikeApiResponse.resources[0];
+  }
+
+  protected async validateRequest(
+    payload: ResponseActionsClientWriteActionRequestToEndpointIndexOptions
+  ): Promise<ResponseActionsClientValidateRequestResponse> {
+    // TODO:PT support multiple agents
+    if (payload.endpoint_ids.length > 1) {
+      return {
+        isValid: false,
+        error: new ResponseActionsClientError(
+          `[body.endpoint_ids]: Multiple agents IDs not currently supported for Crowdstrike`,
+          400
+        ),
+      };
+    }
+
+    return super.validateRequest(payload);
+  }
+
+  async isolate(
+    actionRequest: IsolationRouteRequestBody,
+    options: CommonResponseActionMethodOptions = {}
+  ): Promise<ActionDetails> {
+    const reqIndexOptions: ResponseActionsClientWriteActionRequestToEndpointIndexOptions = {
+      ...actionRequest,
+      ...this.getMethodOptions(options),
+      command: 'isolate',
+    };
+
+    if (!reqIndexOptions.error) {
+      let error = (await this.validateRequest(reqIndexOptions)).error;
+
+      if (!error) {
+        try {
+          await this.sendAction(SUB_ACTION.HOST_ACTIONS, {
+            ids: actionRequest.endpoint_ids,
+            command: 'contain',
+          });
+        } catch (err) {
+          error = err;
+          // throw new Error(err);
+        }
+      }
+
+      reqIndexOptions.error = error?.message;
+
+      if (!this.options.isAutomated && error) {
+        throw error;
+      }
+    }
+
+    const actionRequestDoc = await this.writeActionRequestToEndpointIndex(reqIndexOptions);
+
+    await this.updateCases({
+      command: reqIndexOptions.command,
+      caseIds: reqIndexOptions.case_ids,
+      alertIds: reqIndexOptions.alert_ids,
+      actionId: actionRequestDoc.EndpointActions.action_id,
+      hosts: actionRequest.endpoint_ids.map((agentId) => {
+        return {
+          hostId: agentId,
+          hostname: actionRequestDoc.EndpointActions.data.hosts?.[agentId].name ?? '',
+        };
+      }),
+      comment: reqIndexOptions.comment,
+    });
+
+    if (!actionRequestDoc.error) {
+      await this.writeActionResponseToEndpointIndex({
+        actionId: actionRequestDoc.EndpointActions.action_id,
+        agentId: actionRequestDoc.agent.id,
+        data: {
+          command: actionRequestDoc.EndpointActions.data.command,
+        },
+      });
+    }
```

Comment:
Remove this - unless you know for sure that the response action was completed. My guess is you don't know that and we'll have to implement logic for the background task to check on it and then complete it. The response action should remain in `pending` on our side until we can determine that.